_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,647 @@
+
+After a filter is applied to a set of documents to be searched, the algorithm decides whether to perform pre-filtering for an exact kNN search or modified post-filtering for approximate search. The approximate search with filtering guarantees the top number of closest vectors in result.
+
+Lucene also provides the capability to operate its `KnnVectorQuery` over a subset of documents. To learn more about Lucene’s new capability, see the [Apache Lucene Documentation](https://issues.apache.org/jira/browse/LUCENE-10382).
+
+### Filtered search performance
+
+Filtering that is tightly integrated with the Lucene HNSW algorithm implementation allows you to to apply k-NN searches more efficiently both in terms of relevancy of search results and performance. Consider that if you do an exact search on a large data set, the results are slow, and post-filtering does not guarantee the required number of results you specify for the `k` value.
+With this new capability, you can create an approximate k-NN search and apply filters, with the amount of results you need.
+
+The following workflow diagram shows how the HSNW algorithm decides which type of filtering to apply to a search based on the volume of documents, and number of `k` points in the index that you search with a filter. The variables shown in the diagram are described in the table below.
+
+Variable | Description |
+-- | -- | -- |
+N | Number of documents in the index.
+P | Number of documents in the search set after the filter is applied using the formula: P <= N.
+q | The search vector.
+k | The maximum number of vectors to return in the response.
+
+![How the algorithm evaluates a doc set]({{site.url}}{{site.baseurl}}/images/hsnw-algorithm.png)
+
+***Figure 1: Filter algorithm workflow***
+
+## Filter approaches by use case
+
+Depending on the data set that you are searching, you might choose a different approach to minimize recall or latency. You can create filters that are either: very selective (80%), somewhat selective (38%), or not very selective (2.5%). The selectiveness percentage indicates the amount the filter returns for any given document set in an index.
+
+#### Filter selectiveness with latency per doc set volume
+
+Number of vectors | Selectiveness of filter, % | k | Recall | Latency
+-- | -- | -- | -- | --
+10M | 2.5 | 100 | score_script | score_script
+10M | 38 | 100 | lucene_filtering | Boolean filter
+10M | 80 | 100 | score_script | lucene_filtering
+1M | 2.5 | 100 | lucene_filtering | score_script
+1M | 38 | 100 | lucene_filtering | lucene_filtering / score_script
+1M | 80 | 100 | Boolean filter | lucene_filtering
+
+In this context, `score_script` is essentially a brute force search, whereas a Boolean filter is an approximate k-NN search with post-filtering.
```

Comment:
maybe we need to change `score_script` to Scoring script, this way both Scoring script and Boolean filter will look identical. Essentially in this context we want to refer both in the same manner.